source/tests/SGM/SGMGratingVelocityTests.cpp

```diff
@@ -0,0 +1,263 @@
+#include "SGMGratingVelocityTests.h"
+
+#include "beamline/AMPVControl.h"
+#include <QDebug>
+#include "actions3/AMListAction3.h"
+#include "actions3/AMActionSupport.h"
+#include "beamline/AMControlSet.h"
+#include "actions3/actions/AMWaitAction.h"
+#include "beamline/CLS/CLSMAXvMotor.h"
+#include "beamline/SGM/energy/SGMGratingTranslationStepControl.h"
+GratingVelocityTestConditions::GratingVelocityTestConditions(double encStart, double encEnd, double gratingTransTarget, double velo, double veloBase, double accel)
+{
+	encoderStart = encStart;
+	encoderEnd = encEnd;
+	gratingTranslationTarget = gratingTransTarget;
+	velocity = velo;
+	velocityBase = veloBase;
+	acceleration = accel;
+}
+
+SGMGratingVelocityTests::SGMGratingVelocityTests(QObject* parent) :
+	QObject(parent)
+{
+	gratingAngleEncoder_ =  new AMPVwStatusControl("Grating Angle",
```

Comment:
Sorry, I thought I'd caught all these. The old version of QtCreator on SL5.4 mangles lines like this, even if I set the indent options to the same as my laptop :/